packages/react/modules/components/adapter-context/adapter-context.ts

```diff
@@ -23,5 +23,22 @@ const createAdapterContext = (
 const initialAdapterContext = createAdapterContext();
 const AdapterContext = React.createContext(initialAdapterContext);
 
+const selectAdapterConfigurationStatus = (
+  configurationStatus: TAdapterConfigurationStatus
+) => ({
+  isReady: configurationStatus === TAdapterConfigurationStatus.Configured,
+  isUnconfigured:
+    configurationStatus === TAdapterConfigurationStatus.Unconfigured,
+  isConfiguring:
+    configurationStatus === TAdapterConfigurationStatus.Configuring,
+  isConfigured: configurationStatus === TAdapterConfigurationStatus.Configured,
+});
+
+const useAdapterContext = () => React.useContext(AdapterContext);
+
 export default AdapterContext;
-export { createAdapterContext };
+export {
+  createAdapterContext,
+  useAdapterContext,
+  selectAdapterConfigurationStatus,
```

Comment:
Can we combine the context and the selector?

```ts
const { isConfigured } = useAdapterConfigurationStatus(configurationStatus)
```

The `useAdapterContext` is then used internally in the `useAdapterConfigurationStatus` hook

Reply:
Yeah thought about stuff like that but it got too late (brain fried). I also had to allow just passing in the selector `useAdapterContext(selectConfigurationStatus)` as in Redux.

Reply:
The ideal goal would be to have the selectors for Redux and Broadcast the same. Not even shared maybe but similar to ease things.